### PR TITLE
Add `toBuilder()` to Barbershop

### PR DESCRIPTION
**Pull request by notmikedavis**
This will allow us to construct a Barbershop with new templates without reloading all existing templates from the database

### CODE REVIEW

File: barber/src/main/kotlin/app/cash/barber/Barbershop.kt [hunk 1]
```diff
@@ -50,6 +50,9 @@ interface Barbershop {
   /** Get any Warnings raised from initial install and validation */
   fun getWarnings(): List<String>
 
+  /** Transform Barbershop back into a builder with all Templates installed */
```
Comment (adrw) [inline barber/src/main/kotlin/app/cash/barber/Barbershop.kt:53]:
```suggestion
  /** Create a new Builder from the existing Barbershop that includes all existing templates and documents. */
```

File: barber/src/main/kotlin/app/cash/barber/Barbershop.kt [hunk 2]
```diff
@@ -50,6 +50,9 @@ interface Barbershop {
...
+  fun toBuilder(): BarbershopBuilder
```
Comment (adrw) [inline barber/src/main/kotlin/app/cash/barber/Barbershop.kt:54]:
```suggestion
  fun newBuilder(): BarbershopBuilder
```

**Review by adrw — APPROVED** [review]

File: barber/src/main/kotlin/app/cash/barber/Barbershop.kt (hunk 1 above)
Comment (notmikedavis) [reply barber/src/main/kotlin/app/cash/barber/Barbershop.kt]: Done

File: barber/src/main/kotlin/app/cash/barber/Barbershop.kt (hunk 2 above)
Comment (notmikedavis) [reply barber/src/main/kotlin/app/cash/barber/Barbershop.kt]: Done

**Pull request merged**